### DISCUSSION TITLE
Passing on received message object

### DISCUSSION THREAD

**Discussion by joaoroscoe** [alexandrainst/node-red-contrib-postgresql#51] [Q&A] [answered]
I have several situations in which the code that will process the result of a query also needs to use data from the message object that was consumed by the postgresql node. To handle that, I use junction nodes. Would it be possible passing on the content of the received object, along with the result of the query - perhaps using a check box in the configuration ?
I'd be willing to try, especially if I could get some guidance...

**Comment by Alkarex** [comment, marked as answer]
I am not sure of your exact use-case, but it is pretty standard for Node-RED nodes to use `msg.payload` for the main input and output.
If you need to persist some information, you can always put a copy in `msg.somethingElse`, which should be available even after passing through node-red-contrib-postgresql.

**Reply by joaoroscoe** [reply]
Now, this was **one heck of a killer tip** for a node red newbie such as myself. Thank you very much for your time, and for your patience.
